ni_measurementlink_service/_internal/grpc_servicer.py

```diff
@@ -83,7 +83,28 @@ def time_remaining(self) -> float:
 
     def abort(self, code: grpc.StatusCode, details: str) -> None:
         """Aborts the RPC."""
-        self._grpc_context.abort(code, details)
+        try:
+            self._grpc_context.abort(code, details)
+        except Exception as e:
+            if isinstance(e, Exception):
+                raise CustomRpcError(code)
+            raise e
```

Comment:
```suggestion
        except Exception as e:
            # If gRPC raised an empty Exception, replace it with an RpcError so our logging interceptors can query the code/details.
            if e == Exception():
                raise CustomRpcError(code, details)
            raise
```

Reply:
One more thing: use `from e` or `from None` to control how chained exceptions are displayed.

Using `from e` would display "The above exception was the direct cause..."
Using `from None` would just display the CustomRpcError.
I'm on the fence about which is better here.

Doing nothing would display "During handling of the above exception, another exception occurred: ...".

```python
raise CustomRpcError(code, details) from e
```